Mathlib/NumberTheory/Fermat.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2024 Moritz Firsching. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Firsching
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset
+import Mathlib.Algebra.Order.Ring.Basic
+import Mathlib.Algebra.Order.Star.Basic
+import Mathlib.Data.Nat.Prime.Defs
+import Mathlib.Tactic.Ring.RingNF
+
+/-!
+# Fermat numbers
+
+The Fermat numbers are a sequence of natural numbers defined as $2^{2^n} + 1$, for all natural
+numbers $n$.
+
+## Main theorems
+  `coprime_fermat_fermat`: two distinct fermat numbers are coprime.
+-/
+open Nat
+open scoped BigOperators
+
+/-- The Fermat numbers:
+the $n$-th Fermat number $F_n$ is defined as $2^{2^n} + 1$. -/
+def fermat : ℕ → ℕ := fun n ↦ 2 ^ (2 ^ n) + 1
+
+@[simp]
+theorem fermat_zero : fermat 0 = 3 := by rfl
+
+@[simp]
+theorem fermat_one : fermat 1 = 5 := by rfl
+
+@[simp]
+theorem fermat_two : fermat 2 = 17 := by rfl
```

Comment:
Since these are pretty trivial lemmas I think it's fine to have them grouped together like this:
```suggestion
@[simp] theorem fermat_zero : fermat 0 = 3 := rfl
@[simp] theorem fermat_one : fermat 1 = 5 := rfl
@[simp] theorem fermat_two : fermat 2 = 17 := rfl
```